org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/IMultiRootWorkspaceConfigFactory.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2020 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.server;
+
+import java.util.List;
+
+import org.eclipse.lsp4j.WorkspaceFolder;
+import org.eclipse.xtext.workspace.IWorkspaceConfig;
+
+/**
+ * @author Jan Koehnlein - Initial contribution and API
+ * @since 2.11
+ */
+public interface IMultiRootWorkspaceConfigFactory {
```

Comment:
Why did you decide to make this _not_ a subtype of IWorkspaceConfigFactory ?

Reply:
I didn't because these interfaces have no methods in common.
But the API compatibilities you describe suggest to do so.